File: app/MainActivity.java
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RegistroAdapter registroAdapter;
    private List<Registro> registroList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        registroList = new ArrayList<>();
        registroAdapter = new RegistroAdapter(registroList);
        recyclerView.setAdapter(registroAdapter);

        FloatingActionButton fabAddRegistro = findViewById(R.id.fabAddRegistro);
        fabAddRegistro.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Abrir a janela de criação do novo registro
                // Implemente essa lógica de acordo com suas necessidades
            }
        });
    }

    private static class RegistroAdapter extends RecyclerView.Adapter<RegistroAdapter.RegistroViewHolder> {

        private List<Registro> registros;

        public RegistroAdapter(List<Registro> registros) {
            this.registros = registros;
        }

        @Override
        public RegistroViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_registro, parent, false);
            return new RegistroViewHolder(view);
        }

        @Override
        public void onBindViewHolder(RegistroViewHolder holder, int position) {
            Registro registro = registros.get(position);
            holder.tvObra.setText(registro.getObra());
            holder.tvDonoObra.setText(registro.getDonoObra());
            holder.tvEntidadeExecutante.setText(registro.getEntidadeExecutante());
        }

        @Override
        public int getItemCount() {
            return registros.size();
        }

        public static class RegistroViewHolder extends RecyclerView.ViewHolder {
            TextView tvObra;
            TextView tvDonoObra;
            TextView tvEntidadeExecutante;

            public RegistroViewHolder(View itemView) {
                super(itemView);
                tvObra = itemView.findViewById(R.id.tvObra);
                tvDonoObra = itemView.findViewById(R.id.tvDonoObra);
                tvEntidadeExecutante = itemView.findViewById(R.id.tvEntidadeExecutante);
            }
        }
    }

    private static class Registro {
        private String obra;
        private String donoObra;
        private String entidadeExecutante;

        public Registro(String obra, String donoObra, String entidadeExecutante) {
            this.obra = obra;
            this.donoObra = donoObra;
            this.entidadeExecutante = entidadeExecutante;
        }

        public String getObra() {
            return obra;
        }

        public String getDonoObra() {
            return donoObra;
        }

        public String getEntidadeExecutante() {
            return entidadeExecutante;
        }
    }
}

File: app/src/main/java/com/example/visou/MainActivity.kt
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import com.example.visou.R

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var registroAdapter: RegistroAdapter
    private val registros: MutableList<Registro> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Inicializar a RecyclerView
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        registroAdapter = RegistroAdapter(registros)
        recyclerView.adapter = registroAdapter

        // Configurar o botão de adicionar registro
        findViewById<Button>(R.id.btnAddRegistro).setOnClickListener { adicionarRegistro() }

        // Adicionar um registro de exemplo para fins de demonstração
        adicionarRegistro()
    }

    private fun adicionarRegistro() {
        // Criar um novo registro vazio
        val novoRegistro = Registro("", "", "")
        registros.add(novoRegistro)
        registroAdapter.notifyDataSetChanged()
    }
}

class RegistroAdapter(private val registros: List<Registro>) :
    RecyclerView.Adapter<RegistroAdapter.RegistroViewHolder>() {

    class RegistroViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val etObra: EditText = view.findViewById(R.id.etObra)
        val etDonoObra: EditText = view.findViewById(R.id.etDonoObra)
        val etEntidadeExecutante: EditText = view.findViewById(R.id.etEntidadeExecutante)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RegistroViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_registro, parent, false)
        return RegistroViewHolder(view)
    }

    override fun onBindViewHolder(holder: RegistroViewHolder, position: Int) {
        val registro = registros[position]
        holder.etObra.setText(registro.obra)
        holder.etDonoObra.setText(registro.donoObra)
        holder.etEntidadeExecutante.setText(registro.entidadeExecutante)
    }

    override fun getItemCount(): Int {
        return registros.size
    }
}

data class Registro(
    val obra: String,
    val donoObra: String,
    val entidadeExecutante: String
)
